tests/integrations/sklearn/test_sklearn_api.py

```diff
@@ -0,0 +1,129 @@
+import pytest
+import numpy as np
+
+from sklearn import svm, datasets
+from sklearn.naive_bayes import GaussianNB
+from sklearn.isotonic import IsotonicRegression
+from sklearn.decomposition import PCA
+
+
+from sklearn.utils.validation import check_is_fitted
+
+from hyperactive.integrations import HyperactiveSearchCV
+from hyperactive.optimizers import RandomSearchOptimizer
+
+
+iris = datasets.load_iris()
+X, y = iris.data, iris.target
+
+
+ir = IsotonicRegression()
+nb = GaussianNB()
+svc = svm.SVC()
+pca = PCA(n_components=2)
+
+
+parameters = {"kernel": ["linear", "rbf"], "C": [1, 10]}
+opt = RandomSearchOptimizer()
+
+
+def test_fit():
```

Comment:
these tests (this and below) feel a bit redundant, I think this should already be checked by `parametrize_with_checks`, although I cannot tell for sure for all the things you are testing.

Reply:
The test_sklearn_api-file tests the methods and attributes of our public API. This might also include properties that are not covered by `parametrize_with_checks` in the future.

Aside from that I am not opposed to some redundancy in my tests, since it can increase test-safety according to [this model](https://en.wikipedia.org/wiki/Swiss_cheese_model).